### ISSUE TITLE
[TheiaProk wfs] upgrade StxTyper version when available & ensure all OPERON output types are captured appropriately

### ISSUE THREAD

**Issue by kapsakcj** [theiagen/public_health_bioinformatics#694]
:cool:

### request 1 - upgrade version

This is an internal request from me, but would be desired by our public health partners (CA) and users. It is a low priority request at the moment but may be upgraded priority pending on feedback from CA partners

Version of StxTyper in theiaprok workflows currently: **1.0.24**

Current version available on NCBI StxTyper GitHub: **1.0.27**

~~FYI in the coming months there will be a new version released with minor bug fixes for edge cases. I will reply back when it's available.~~

~~StaPH-B/Curtis will make a docker image available for the new version when it is released.~~ We can copy this to our google artifact registry once available.

### request 2 - update parsing of results; potentially add new string output to task and workflows

I want to also use this opportunity to ensure that the stxtyper task in theiaprok accurately captures all potential OPERON types.

I accidentally omitted searching for hits with `EXTENDED` and `AMBIGUOUS` operon type so that is not very visible to the user unless they look in the stxtyper_report TSV file (raw output from the tool). They are rarely found, but still would be good to capture and notify the end user of the TheiaProk workflow in case they are found by stxtyper.

From StxTyper README description:

> EXTENDED The coding sequence extends beyond the reference stop codon for one or both of the reference proteins

and

> AMBIGUOUS StxTyper found an ambiguous base in the query sequence (e.g., N), this could be the result sequencing or assembly error so the user might want to take a closer look at the sequence.

**Comment by kapsakcj** [comment]
The stxtyper v1.0.27 docker image is available from staphb: https://github.com/StaPH-B/docker-builds/tree/master/build-files/stxtyper/1.0.27

the docker image should be copied over to our google artifact registry and the stxtyper WDL task should be updated to use the image as default

**Comment by kapsakcj** [comment]
Perhaps hold off on this update.

It looks like a new stxtyper version is incoming. https://github.com/ncbi/stxtyper/tree/dev

Let's wait until that is released and a new docker image can be created via StaPH-B

**Comment by kapsakcj** [comment]
New version is available. I can make a staphb docker image

https://github.com/ncbi/stxtyper/releases/tag/v1.0.31

**Comment by kapsakcj** [comment]
OK docker image has been copied to our GAR `us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.31`.

This issue is ready to be worked on
